mc-sema/bin_descend/get_cfg.py

```diff
@@ -449,6 +449,54 @@ def isExternalData(fn):
         return False
 
 
+def sanityCheckJumpTableSize(table_ea, ecount):
+    """ IDA doesn't correctly calculate some  jump table sizes """
+    if not isLinkedElf():
+        return ecount
+
+    if getBitness() != 64:
+        return ecount
+
+    table_insn = idautils.DecodeInstruction(table_ea)
+    if table_insn is None:
+        DEBUG("Could not decode instruction at {:x}\n".format(table_insn))
+        return ecount
+
+    if table_insn.Operands[0].type != idc.o_reg:
+        return ecount
+
+    DEBUG("Sanity checking table at {:x}\n".format(table_ea))
+
+    # get register we jump with
+    jmp_reg = table_insn.Operands[0].value
+
+    inst_ea = table_ea
```

Comment:
OK, so adding this comment would be useful to show the pattern being recognized:

```assembly
  cmp eax, num_entries
  jg bad_entry
fall_through:
  mov rax, qword [raw * ptr_size + table_base_address]
  jmp rax
bad_entry:
  ...
```